### PR TITLE
aws - filter to check the state of securityhub

### LINKED ISSUE #5852: Enable/Disable SecurityHub
**Is your feature request related to a problem? Please describe.**
I want to check if SecurityHub is enabled or not for all the accounts and take an action accordingly

**Describe the solution you'd like**
We can create a filter and action on `resource: aws.account`. Filter uses api `describe_hub`(https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/securityhub.html#SecurityHub.Client.describe_hub)
If we get a response back => sechub is enabled. If we get an exception, take action (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/securityhub.html#SecurityHub.Client.enable_security_hub)

**Describe alternatives you've considered**
Create a new resource type

**Additional context**
NA

### PR DESCRIPTION
**Pull request by PratMis**
Closes #5852

### CODE REVIEW

File: c7n/resources/account.py [hunk 1]
```diff
@@ -1731,3 +1731,76 @@ def process(self, resources):
...
+
+    :example:
+
+    .. code-block:: yaml
+
+       policies:
+         - name: check-securityhub-status
+           resource: aws.account
+           filters:
+            - type: sechub-enabled
```
Comment (kapilt) [inline c7n/resources/account.py:1748]: re filter name, how about securityhub, with a parameter on enabled: [boolean]

File: c7n/resources/account.py [hunk 2]
```diff
@@ -1731,3 +1731,76 @@ def process(self, resources):
...
+
+    schema = type_schema('sechub-enabled', state={'type': 'boolean'})
+
+    def process(self, resources, event=None):
+        state = self.data.get('state', True)
+        client = local_session(self.manager.session_factory).client('securityhub')
+        try:
+            sechub = client.describe_hub()
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'InvalidAccessException':
```
Comment (kapilt) [inline c7n/resources/account.py:1763]: this needs a comment for the case it represents (region not activated)

minor, this try/except can be replaced with a call to self.manager.retry with the ignorable error codes.

File: c7n/resources/account.py [hunk 3]
```diff
@@ -1731,3 +1731,76 @@ def process(self, resources):
...
+                sechub = None
+            else:
+                raise
+        if state == bool(sechub):
+            return resources
+        return []
+
+
+@actions.register('set-sechub')
+class SetSechub(BaseAction):
```
Comment (kapilt) [inline c7n/resources/account.py:1773]: please remove this action, it doesn't represent best practices usages for a security-hub

File: c7n/resources/account.py (hunk 1 above)
Comment (PratMis) [reply c7n/resources/account.py]: Sounds good, addressed!

File: c7n/resources/account.py (hunk 3 above)
Comment (PratMis) [reply c7n/resources/account.py]: Dropped the action as per your suggestion

File: c7n/resources/account.py (hunk 2 above)
Comment (PratMis) [reply c7n/resources/account.py:1764]: Wrapped self.manager.retry inside try/except to make sure we catch `InvalidAccessException`. Thanks for the suggestion!

File: tests/data/placebo/test_enable_securityhub/securityhub.DescribeHub_1.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "status_code": 401,
+    "data": {
+        "Error": {
+            "Message": "AccountId: 644160558196 is not enabled for securityhub. Currentstate: false",
```
Comment (PratMis) [inline tests/data/placebo/test_enable_securityhub/securityhub.DescribeHub_1.json:5]: This test was executed in a different region where security is not enabled

File: c7n/resources/account.py (hunk 3 above)
Comment (kapilt) [reply c7n/resources/account.py]: just to clarify security hub enablement is typically a multi account orchestration to setup correctly for aggregation, its closer to something we would want to use c7n_guardian for enablement. we probably want to have the enablement for it there.

Comment (PratMis) [reply c7n/resources/account.py]: thanks and noted. I'll take a look at c7n_guardian and see if I can add this

File: c7n/resources/account.py (hunk 2 above)
Comment (kapilt) [reply c7n/resources/account.py:1764]: so the latest versions of trunk these error codes can be given directly to retry, instead of a try/except.

Comment (PratMis) [reply c7n/resources/account.py]: ah, got it! Didn't know about that, modified code to use that. Thanks!

**Pull request merged**